### PR TITLE
Enhancement to Dimension and TensorShape API documentation

### PR DESCRIPTION
**Pull request by SSaishruthi**
This PR is intended to help beginners understand the functionality of both Dimension and TensorShape API better by providing additional notes about each function and making it self-explanatory.

### CODE REVIEW

**Comment by SSaishruthi** [comment]
I signed it!

**Comment by rthadur** [comment]
> I signed it!

@SSaishruthi we did not see CLA , can you please make sure you use same username and email while signing CLA

**Comment by SSaishruthi** [comment]
CLA issue has been resolved.

**Comment by SSaishruthi** [comment]
Hi @rthadur,
Is this PR in review? 
TIA

**Comment by rthadur** [comment]
> Hi @rthadur,
> Is this PR in review?
> TIA

yes, can you please resolve conflicts.

**Comment by SSaishruthi** [comment]
@rthadur 
Resolved

File: tensorflow/python/framework/tensor_shape.py [hunk 1]
```diff
@@ -329,18 +379,18 @@ def __add__(self, other):
...
+      other: Another Dimension.
 
     Returns:
       A Dimension whose value is the sum of `self` and `other`.
     """
     return self + other
 
   def __sub__(self, other):
-    """Returns the subtraction of `other` from `self`.
+    """Subtraction of `other` from `self`.
```
Comment (qlzh727) [inline tensorflow/python/framework/tensor_shape.py]: This actually gives user the wrong impression that the value of self is being updated, which is not in this case. The method only return the subtracted value without updating self and other.

Comment (qlzh727) [reply tensorflow/python/framework/tensor_shape.py]: same of all the methods above and below.

File: tensorflow/python/framework/tensor_shape.py [hunk 2]
```diff
@@ -799,16 +875,21 @@ def rank(self):
...
+  
   @property
   def ndims(self):
     """Deprecated accessor for `rank`."""
     return self.rank
 
   def __len__(self):
-    """Returns the rank of this shape, or raises ValueError if unspecified."""
+    """Obtain rank of the shape.
+    Returns:
```
Comment (qlzh727) [inline tensorflow/python/framework/tensor_shape.py]: should have a blank line between docstring and return section.

File: tensorflow/python/framework/tensor_shape.py [hunk 3]
```diff
@@ -821,7 +902,13 @@ def __bool__(self):
   __nonzero__ = __bool__
 
   def __iter__(self):
-    """Returns `self.dims` if the rank is known, otherwise raises ValueError."""
+    """
```
Comment (qlzh727) [inline tensorflow/python/framework/tensor_shape.py]: Not sure why the docstring is empty.

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Changes for the change. Honestly, I don't think the currently PR increase the readability a lot since most of the method and code are self-explanatory. The comment and doc is usually needed when the code has extra complexity. If the code itself can already serve as doc, then adding more doc will actually reduce the readability since it provide more noisy.

I will also let Mark from doc team to do further review.

**Comment by SSaishruthi** [comment]
Working on the changes.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**